data/fin.lean

```diff
@@ -88,3 +91,30 @@ by cases i; refl
   {n} {C : fin (succ n) → Sort*} {H0 Hs} (i : fin n) :
   @fin.cases n C H0 Hs i.succ = Hs i :=
 by cases i; refl
+
+def nat_expand_type {n} (i : fin n) (k) : fin (n + k) :=
+⟨i.1,
+begin
+  have h : i.val < n, by apply i.2,
+  apply lt_add_of_lt_of_nonneg,
+  apply h,
+  have h2 : k = 0 ∨ k > 0, by apply nat.eq_zero_or_pos,
+  rw or_comm at h2,
+  rw le_iff_lt_or_eq,
+  rw eq_comm,
+  apply h2,
+end
+⟩
+
+def contract_type_first {n m} (i : fin (n + m)) (h : i.val < n) : fin (n) :=
+⟨i.1, begin apply h end⟩
+
+def contract_type_second {n m} (i : fin (n + m)) (h: i.val >= n) : fin (m) :=
+⟨i.1 - n,
+  begin
+    have h2: i.val < n + m, by apply i.2,
+    rw nat.sub_lt_right_iff_lt_add h,
+    simp,
+    apply h2,
+  end
```

Comment:
Can the whole proof be replaced with `simp [nat.sub_lt_right_iff_lt_add h,i.2]`?

Reply:
Yes.